Add a guide to dynamic Kubernetes registration

ptgott opened the pull request:
Closes #17629

ptgott committed:
Add a guide to dynamic Kubernetes registration

Closes #17629

tigrato commented on docs/pages/kubernetes-access/guides/dynamic-registration.mdx:
```diff
@@ -0,0 +1,458 @@
+  allow:
+    rules:
+    - resources:
+      - kube_cluster
```
Can we update this section as well: 
https://goteleport.com/docs/access-controls/reference/#roles

In the meanwhile, if possible please correct #16247

tigrato commented on docs/pages/kubernetes-access/guides/dynamic-registration.mdx:
````diff
@@ -0,0 +1,458 @@
+```yaml
+kind: kube_cluster
+version: v3
+metadata:
````
Maybe it makes sense to mention that we can add labels and dynamic labels to the configuration here. Useful for Teleport RBAC

ptgott committed:
Address PR feedback

- Add to and edit the list of dynamic resources
  Closes #16247

- Describe static and dynamic labels in the `kube_cluster` resource

alexfornuto commented on docs/pages/kubernetes-access/guides.mdx:
```diff
@@ -4,7 +4,15 @@ description: Detailed guides for configuring Teleport Kubernetes Access.
+- [Teleport Kubernetes Access and Trusted Clusters](./guides/federation.mdx):
+  Federated Access using Teleport Trusted Clusters.
+- [Kubernetes Access Multiple Clusters](./guides/multiple-clusters.mdx):
+  Connecting Multiple Clusters with Teleport Kubernetes Access.
+- [Kubernetes Access from a Standalone Teleport
+  Cluster](./guides/standalone-teleport.mdx): Connecting a Teleport cluster to a
```
Not part of your change set, but something I noticed; these two link descriptions use a different tense than the others. What about:

```suggestion
  Connect Multiple Clusters with Teleport Kubernetes Access.
- [Kubernetes Access from a Standalone Teleport
  Cluster](./guides/standalone-teleport.mdx): Connect a Teleport cluster to a
```

alexfornuto commented on docs/pages/kubernetes-access/guides/dynamic-registration.mdx:
```diff
@@ -0,0 +1,514 @@
+
+Install the Teleport Kubernetes Service on your Linux host:
+
+(!docs/pages/includes/install-linux.mdx!)
```
Does this include need to be scoped so that enterprise users get the enterprise installation instructions? Or do enterprise users not need the enterprise binary for this service, even when connecting to an enterprise cluster?

ptgott replied:
`install-linux.mdx` uses `ScopedBlocks` to show Enterprise/OSS content

alexfornuto commented on docs/pages/kubernetes-access/guides/dynamic-registration.mdx:
````diff
@@ -0,0 +1,514 @@
+Teleport Cloud tenant and `TOKEN` to the join token we created earlier:
+
+```code
+$ PROXY_SERVICE=teleport.example.com:443;
+$ TOKEN=(=presets.tokens.first=);
````
I had a different take on using shell vars like this in comments below that I made earlier. Now that we have the in-page edits feature documented, could/should we use them instead?

ptgott replied:
I've used in-page edits, but also kept the environment variables so we can include comments above them with a suggested value.

alexfornuto commented on docs/pages/kubernetes-access/guides/dynamic-registration.mdx:
````diff
@@ -0,0 +1,514 @@
+```diff
+   roles:
+   - access
+   - auditor
````
```suggestion
   - auditor
   - kube-user
```

Since it should be there from the previous step.

alexfornuto commented on docs/pages/kubernetes-access/guides/dynamic-registration.mdx:
```diff
@@ -0,0 +1,514 @@
+After creating resources, `get-kubeconfig.sh` writes a new kubeconfig to a file
+called `kubeconfig` in the directory where you run the script.
+
+Execute the following command:
```
```suggestion
Run the `get-kubeconfig.sh` script:
```

Just to be more descriptive.

alexfornuto reviewed (approved):
👍, pending outstanding comments.

ptgott commented on docs/pages/kubernetes-access/guides/dynamic-registration.mdx:
```diff
@@ -0,0 +1,458 @@
+$ PROXY_SERVICE=teleport.example.com:443;
+$ TOKEN=(=presets.tokens.first=);
+$ sudo teleport configure \
+--proxy=${PROXY_SERVICE?} \
+--roles=kube \
+--token=${TOKEN?} \
```
These are to cause the shell to print an error if the variables are unset, which prevents users who copy the snippet from accidentally assigning empty values

ptgott commented on docs/pages/kubernetes-access/guides/dynamic-registration.mdx:
````diff
@@ -0,0 +1,458 @@
+Retrieve your user:
+
+```code
+$ TELEPORT_USER=myuser
+$ tctl get user/${TELEPORT_USER?} > user.yaml
````
I wanted to make the need to assign a value more prominent. I've suggested using our new `Var` components instead.

ptgott committed:
Respond to PR feedback
